src/ui/public/url/url.js

```diff
@@ -114,7 +115,11 @@ export function KbnUrlProvider($injector, $location, $rootScope, $parse, Private
 
       // if evaluation can't be made, throw
       if (_.isUndefined(p)) {
-        throw new Error('Replacement failed, unresolved expression: ' + expr);
+        throw new Error(
+          i18n.translate('common.ui.url.replacementFailedErrorMessage', {
```

Comment:
The same as above, please inject`i18n` service https://github.com/elastic/kibana/blob/7b27cbe0282ebf32d932fc6bdc561d8148586f68/src/ui/public/url/url.js#L30